pkg/grizzly/workflow.go

```diff
@@ -231,10 +231,14 @@ func Apply(registry Registry, resources Resources) error {
 
 		existingResource, err := handler.GetRemote(resource)
 		if errors.Is(err, ErrNotFound) {
-			uid := resource.GetSpecString("uid")
-			if uid != resource.Name() {
-				return fmt.Errorf("uid '%s' and name '%s', don't match", uid, resource.GetSpecString("uid"))
+			// synthetic monitoring doesn't have uid in the spec
+			if resource.Kind() != "SyntheticMonitoringCheck" {
```

Comment:
We need to fake a hidden somehow, so that every resource has one.

I have made it with `type.job`. We could make grizzly fail if the name does not match the type.job in the spec.

That way workflow.go can be the same across all handlers.

Reply:
how about we check the uid exist or not, if exist, we check it is the same as type.job, otherwise ignore it. to have a fake uid would have to keep the uid updated when check method changes etc, and also i need to marshal/unmarshal the spec to get check object, it would be less evident to be maintained.